jobqueue/src/main/java/com/path/android/jobqueue/network/NetworkUtilImpl.java

```diff
@@ -24,16 +27,43 @@ public void onReceive(Context context, Intent intent) {
                 // through getActiveNetworkInfo() or getAllNetworkInfo().
                 listener.onNetworkChange(isConnected(context));
             }
-        }, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+        }, getNetworkIntentFilter());
     }
 
     @Override
     public boolean isConnected(Context context) {
+        if (isDozing(context)) {
+            return false;
+        }
+
         ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo netInfo = cm.getActiveNetworkInfo();
         return netInfo != null && netInfo.isConnectedOrConnecting();
     }
 
+    @TargetApi(23)
+    private static IntentFilter getNetworkIntentFilter() {
+        IntentFilter networkIntentFilter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
+        if (VERSION.SDK_INT >= 23) {
```

Comment:
I agree with this.

I believe I had Lint issues or Android Studio yelling at me about VERSION_CODES.M because of the compileSdkVersion or targetSdkVersion being lower than 23; could be making that up haha.